specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+  HRESULT remove_RenderAdapterLUIDChanged(
+      [in] EventRegistrationToken token);
+}
+/// This is the interface that handles texture streaming.
+[uuid(afca8431-633f-4528-abfe-7fc3bedd8962), object, pointer_default(unique)]
+interface ICoreWebView2StagingTextureStream : IUnknown {
+  /// Get the stream ID of the object that is used when calling CreateTextureStream.
+  /// The caller must free the returned string with CoTaskMemFree. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  // MSOWNERS: TBD (wv2core@microsoft.com)
+  [propget] HRESULT StreamId([out, retval] LPWSTR* id);
+  /// Adds an allowed url origin for the given stream id. The stream requests
+  /// could be made from any frame, including iframes, but these origins
+  /// must be registered first in order for the request to succeed.
+  /// The added filter will be persistent until
+  /// ICoreWebView2StagingTextureStream is destroyed or
+  /// RemoveAllowedOrigin is called.
+  /// The renderer does not support wildcard so it will compare
+  /// literal string input to the requesting frame origin. So, the input string
+  /// should have a scheme like https://.
+  /// For example, https://www.valid-host.com, http://www.valid-host.com are
+  /// valid origins but  www.valid-host.com, or *.valid-host.com. are not
+  /// valid origins.
+  /// getTextureStream() will fail unless the requesting frame's origin URL is
+  /// added to the request filter.
+  HRESULT AddAllowedOrigin([in] LPCWSTR origin);
+  /// Remove added origin, which was added by AddAllowedOrigin.
+  HRESULT RemoveAllowedOrigin([in] LPCWSTR origin);
+  /// Listens for stream requests from the Javascript's getTextureStream call
+  /// for this stream's id. It is called for the first request only, and will
+  /// not be called with subsequent requests of same stream id from any pages
+  /// in the middle of request handling or after it returns success.
+  /// The request is regarded as success only when the host provides the stream,
+  /// Present API call, within 10s after being requested.
+  HRESULT add_StartRequested(
+      [in] ICoreWebView2StagingTextureStreamStartRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_StartRequested(
+      [in] EventRegistrationToken token);
+  /// Listen to stop stream request once the stream started.
+  /// It is called when user stop all streaming requests from
+  /// the renderers (Javascript) or the host calls the Stop API. The renderer
+  /// can stream again by calling the streaming request API.
+  /// The renderer cleared all registered buffers before sending
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  HRESULT add_StopRequested(

```

Comment:
(I'm not sure i might already replied on other comment. somehow view doesn't work well).

The host is a streaming provider that means it is the one that decide whether it will start or stop streaming. The browser in this case is just client who can request of it.  

client: Start request means, I'm ready to accept the texture, please send me texture.
host: ok, I will send a texture (Started state) or no I don't want to (Stopped state).

client: Stop request means, I don't want to listen to stream from you so I've already clean any resources for it. you don't have to send me a texture.
host: ok, I will stop sending you (Stopped state), or hmm I want to keep sending you texture (Stopped state) although no one listen to.

It looks like we are in two different worlds so it is easier to think Start/State is from the host side perspective as the API is host face one.